mitodl/sso-django-prototype
# Social auth OIDC integration

#### Pre-Flight checklist


#### What's this PR do?
Creates an index page from which a user can click Login.  This will redirect them to the configured Keycloak instance where they can authenticate, afterwhich they are redirected back to the index page which will display a message that includes their username which was received from Keycloak.

#### How should this be manually tested?

1. Configure Keycloak following the instructions here: https://github.com/mitodl/odl-custom-keycloak/wiki/ODL-Developer-Resources#connecting-local-client-with-local-keycloak-mitxonline
2. Open: http://localhost:8091/ and click login
3. Create a Keycloak account or authenticate with an existing Keycloak account.
4. Verify that you are redirected back to the index page and shown your Keycloak username.

#### Where should the reviewer start?
Have Keycloak up and running locally by following the instructions here: https://github.com/mitodl/odl-custom-keycloak/wiki/ODL-Developer-Resources#keycloak-ootb-setup
